crates/modelardb_common/src/metadata/mod.rs

```diff
@@ -194,10 +195,15 @@ impl MetadataDeltaLake {
         table_name: &str,
         rows: Vec<ArrayRef>,
     ) -> Result<DeltaTable, DeltaTableError> {
-        let table_provider = self.session.table_provider(table_name).await?;
-        let batch = RecordBatch::try_new(table_provider.schema(), rows)?;
+        let table = open_table_with_storage_options(
+            format!("{}/{table_name}", self.location),
+            self.storage_options.clone(),
+        )
+        .await?;
+
+        let batch = RecordBatch::try_new(TableProvider::schema(&table), rows)?;
```

Comment:
Maybe add a comment why `TableProvider::schema(&table)` is used instead of `table.schema()`, I assume it is so [`TableProvider::schema()`](https://docs.rs/deltalake/latest/deltalake/datafusion/catalog/trait.TableProvider.html#tymethod.schema) is called instead of [`DeltaTable.schema()`](https://docs.rs/deltalake/latest/deltalake/struct.DeltaTable.html#method.schema).